src/main/scala/viper/gobra/frontend/info/implementation/typing/ProgramTyping.scala

```diff
@@ -94,10 +86,42 @@ trait ProgramTyping extends BaseTyping { this: TypeInfoImpl =>
     }
   }
 
-  private def hasOldExpression(posts: Vector[PExpression]): Messages = {
+  private val globalVarDeclsSortedByPos: PProgram => Vector[PVarDecl] = {
+    val computation: PProgram  => Vector[PVarDecl] = { p =>
+        val pkg: PPackage = tryEnclosingPackage(p).get
+        // the following may only be called once per package, otherwise wildcard identifiers may be desugared multiple
+        // times, leading to different names being generated on different occasions.
+        val unsortedDecls = p.declarations.collect { case d: PVarDecl => d; case PExplicitGhostMember(d: PVarDecl) => d }
+        unsortedDecls.sortBy { decl =>
+          pkg.positions.positions.getStart(decl) match {
+            case Some(pos) => (pos.line, pos.column)
+            case None => violation(s"Could not find the position information of node $decl.")
+          }
+        }
```

Comment:
Feel free to reject this suggestion, I'm a bit unsure about it myself. One advantage I see is that this make it (more?) explicit that this function can only be called with a `PProgram` that belongs to the current package. I'm not sure whether this is also achieved by `tryEnclosingPackage(p).get`. In particular, does fail with a node not in tree exception as the current package is traversed (as opposed to the tree in which `p` lies?)

```suggestion
        require(tree.root.programs.contains(p))
        // the following may only be called once per package, otherwise wildcard identifiers may be desugared multiple
        // times, leading to different names being generated on different occasions.
        val unsortedDecls = p.declarations.collect { case d: PVarDecl => d; case PExplicitGhostMember(d: PVarDecl) => d }
        unsortedDecls.sortBy { decl =>
          tree.root.positions.positions.getStart(decl) match {
            case Some(pos) => (pos.line, pos.column)
            case None => violation(s"Could not find the position information of node $decl.")
          }
        }
```